### PR TITLE
Containerized PC tools image update.

### PR DESCRIPTION
**Pull request by m-dati**
Related to the already merged PC tools update, same ticket, this PR contains the code used to create the container and update the virtualenv that produced the files with the new tools versions.

Check the procedure in the README file.

- Related ticket: https://progress.opensuse.org/issues/106658
- Needles: none
- Verification run: see in https://github.com/m-dati/os-autoinst-distri-opensuse/tree/master/data/publiccloud/venv/*.txt, the new txt files produced following the README procedure.

### CODE REVIEW

File: tools/pctools/README.md
Comment (asmorodskyi) [inline tools/pctools/README.md:37]:
```suggestion
c) run the container sharing the workdir with the SHARDIR defined in the Dockerfile, without other commands, so that the CMD instruction by default will copy here the files with the new versions:
```

**Review by asmorodskyi — APPROVED** [review]
LGTM , but I would like someone else to check it before merge

File: tools/pctools/README.md
Comment (grisu48) [inline tools/pctools/README.md:4]: It is a good habit to always establish the context of the repository to allow readers without prior background information to get a grasp what's going on here, e.g.


```suggestion
The Public Cloud Tools image is a modified SLES image with a pre-installed and configured set of various cloud related tools intended for testing of Cloud Service Providers (CSP) within openQA. It include various tools including command-line interfaces for Azure, Amazon, OpenStack and Google Compute Platform (GCP). This Dockerfile helps to generate the automatically generate the requirements for those utilities.
```

It is also a courtesy to the reader to always introduce abbreviations when used for the first time (or refer to a glossary).

File: tools/pctools/venv_inst.sh
Comment (grisu48) [inline tools/pctools/venv_inst.sh:16]: Might be a good idea to exit with an error message, WDYT?

```suggestion
[ $# -lt 3 ] && (echo "not enough arguments"; exit 1)
```

File: tools/pctools/venv_inst.sh
Comment (grisu48) [inline tools/pctools/venv_inst.sh:3]:
```suggestion
# create a python virtualenv named as test_$1,
```

File: tools/pctools/venv_inst.sh
Comment (grisu48) [inline tools/pctools/venv_inst.sh:1]: Perhaps it's a good idea to fail the bash script in case of errors in the commands?

```suggestion
#!/bin/bash -e
```

File: tools/pctools/venv_inst.sh
Comment (grisu48) [inline tools/pctools/venv_inst.sh:38]: Not needed if you use the shebang with `/bin/bash -e`

File: tools/pctools/venv_inst.sh
Comment (grisu48) [inline tools/pctools/venv_inst.sh:49]: This has no effect because after this the new bash is anyways terminated (bash script start in a new `bash` and don't affect the outside environment)

File: tools/pctools/venv_inst.sh
Comment (grisu48) [inline tools/pctools/venv_inst.sh:20]: Perhaps giving the resources output file a more descriptive variable name might increase the readability?

```suggestion
OUTPUT=$2
```

File: tools/pctools/venv_inst.sh
Comment (grisu48) [inline tools/pctools/venv_inst.sh:41]:
```suggestion
pip freeze > ../$FILE; 
```

Then you don't need to copy the file over later on, but I leave it to you if you like this or not :-)

File: tools/pctools/Dockerfile
Comment (grisu48) [inline tools/pctools/Dockerfile:16]: Why is here another variable (`$DIRAWS`) being used?

This gets a bit too convoluted. We are using an `ARG`, which contains a bash variable, which is passed as an argument to a bash script ...  
Is this level of complexity really necessary? Following the flow is difficult here.

**Review by grisu48 — COMMENTED** [review]
Good job! I've added some improvement suggestions, feel free to take them or comment on them if they are not applicable :slightly_smiling_face:

File: tools/pctools/venv_inst.sh
Comment (m-dati) [reply tools/pctools/venv_inst.sh:16]: thanks, indeed I already did so, but the main process of the command RUN resulted still ok and the build doesn't stop, because in this way the eventual error occurs in a sub-shell (...) but the parent shell is ok. 
I should convert the command using a normal 'if ... then'.

File: tools/pctools/venv_inst.sh
Comment (m-dati) [reply tools/pctools/venv_inst.sh:49]: if you mean the command `cd ..`, I agree :+1:

File: tools/pctools/venv_inst.sh
Comment (m-dati) [reply tools/pctools/venv_inst.sh:41]: I'd prefer to create the file in the specific folder dedicated to that cloud type so that in the final container you have the correct status set for each type. 
Indeed, then that copy '..' is only needed to collect all the files in a folder that will be shared at runtime, from where we can get all the files our needs.

File: tools/pctools/Dockerfile
Comment (m-dati) [reply tools/pctools/Dockerfile:16]: I was using the variable to create a filename with a proper common template for the copy phase. The same variable is ised  to create a venv dir. But I agree, I can change somethings and reduce parameters use

**Comment by m-dati** [comment]
Please find here the log of a build/run execution after last update: http://pasta.qam.suse.de/mY8zjiei

File: tools/pctools/venv_inst.sh
Comment (grisu48) [reply tools/pctools/venv_inst.sh]: Up to you, both approaches work just fine :slightly_smiling_face:

File: tools/pctools/Dockerfile
Comment (grisu48) [inline tools/pctools/Dockerfile:29]: I would concatenate the commands with `&&` otherwise we continue to run this also if there are failures.  
If here .eg. `pip install virtualenv` fails, the `RUN` will still succeed.

Simply replacing all `;` with `&&` should do the job

File: tools/pctools/venv_inst.sh
Comment (grisu48) [inline tools/pctools/venv_inst.sh:58]: Missing newline at the end

**Review by grisu48 — APPROVED** [review]
Two nitpicks, otherwise LGTM :rocket:

**Review by grisu48 — APPROVED** [review]

**Pull request merged**
